docs/ingest.md

```diff
@@ -0,0 +1,115 @@
+
+We change to using `parentPath` when writing to DynamoDB to benefit from querying with a Global Secondary Index and remove the need to recursively query query items to reach the top of our tree.
+
+### Upsert Archive Folders
+
+[This Lambda](/scala/lambdas/ingest-upsert-archive-folders/) is responsible for creating or updating certain folders within the Preservation System. This step is required as the default behaviour of the ingest mechanism is to place the full package within a nominated folder, ingesting as a child of the root node is not possible within the Preservation System. We also have a requirement to update some folder metadata on subsequent ingests, for example, if the Discovery title or description of a dept/series changes, we’d want to update this within the Preservation System.
+
+Both when creating or updating a folder, identifiers are passed through into the Preservation System’s native Identifier feature. Any field prefixed with `id_`, will create or update the identifier of that type. E.g. id_Code will become the Code identifier within the Preservation System. This has been implemented as the values of identifiers can be dependent on upstream system - court documents have a Cite identifier which other document types do not, court documents also exist in a “case folder” with a custom Code identifier.
+
+### Asset OPEX Creation
+
+```mermaid
+flowchart LR;
+    subgraph Map over each asset
+        1[Task: Find existing assts] --> 2
+        2[Choice: Does asset exist?] --> 3
+        3[Pass: Skip ingesting asset]
+        2 --> 4[Task: Create asset OPEX]
+    end
+```
+
+The Preservation System is expecting an [OPEX package](https://developers.preservica.com/documentation/open-preservation-exchange-opex) to ingest. Our process needs to transform the JSON file we started with into a package of this format. We must also deduplicate as part of our ingest process to prevent storing multiple of the same asset in the Preservation System. As an ingest batch could contain thousands of assets with even more files that need copying, we horizontally scale the processing of this step using a Step Function Map State.
+
+For each asset we first [check to see if it's already in the Preservation System](/scala/lambdas/ingest-find-existing-asset/). If it is, we add the `skipIngest` attribute to our DynamoDB table and ignore the asset when generating the OPEX package. We maintain the asset within our DynamoDB table to allow us to reconcile the new ingest candiate against the existing ingested asset to ensure they are the same - we want to appear idempotent to clients.

```

Comment:
1. MIssing comma after "For each asset"
2. Should "maintain" be "retain", maybe?